File: euler4.go
package main

import (
	"fmt"
	"strconv"
)

func isPalindrome(n int) bool {
	testString := strconv.Itoa(n)
	length := len(testString)

	if length == 1 {
		return false
	}

	for i := 0; i < length/2; i++ {
		if testString[i] != testString[length-1-i] {
			return false
		}
	}

	return true
}

func main() {
	max := 999
	result := 0

	for i := max; i > 0; i-- {
		for j := max; j > 0; j-- {
			r := i * j
			if isPalindrome(r) && r > result {
				result = r
			}
		}
	}

	fmt.Println(result)
}

File: euler4/euler4.py
def isPalindrome(number):
    number = str(number)
    length = len(number)
    halfLen = int(length / 2)

    if (length == 1):
        return False

    i = 0
    while i < halfLen:
        if number[i] != number[length - 1 - i]:
            return False
        i = i + 1
    return True

max = 999
result = 0

for i in range(max, 1, -1):
    for j in range(max, i, -1):
        r = i * j
        if isPalindrome(r) and r > result:
            result = r

print(result)
